src/finufft_core.cpp

```diff
@@ -130,8 +126,9 @@ int setup_spreader_for_nufft(finufft_spread_opts &spopts, FLT eps, finufft_opts
   return ier;
 }
 
-void set_nhg_type3(FLT S, FLT X, finufft_opts opts, finufft_spread_opts spopts,
-                   BIGINT *nf, FLT *h, FLT *gam)
+template<typename T>
+static void set_nhg_type3(T S, T X, finufft_opts opts, finufft_spread_opts spopts,
```

Comment:
Shall we pass by reference spopts?

Reply:
Good catch! I think yes, but I'll double-check.

Reply:
Done.